include/mbgl/util/unique_function.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <functional>
+
+namespace mbgl {
+namespace util {
+
+// C++20 non-copying lambda capture, pending C++23 `move_only_function`
```

Comment:
Maybe we can use something like `https://github.com/zhihaoy/nontype_functional/blob/main/include/std23/move_only_function.h` instead so we can more easily migrate to `std::move_only_function` when we can use C++23.

It's also better tested.

Reply:
I'll give it a try.  Would we add that whole thing as a dependency, or just pull in the one file?

Reply:
If it's not too much hassle I would add it to `vendor/nontype_functional`. Could just be the two files we need.